anchore_engine/services/policy_engine/engine/feeds/grypedb_sync.py

```diff
@@ -73,40 +66,45 @@ class GrypeDBSyncManager:
     lock = threading.Lock()
 
     @classmethod
-    def _get_active_grypedb(cls) -> GrypeDBFeedMetadata:
+    def _query_active_dbs(cls, session: Session) -> List[GrypeDBFeedMetadata]:
         """
-        Returns active grypedb instance from db. Raises NoActiveGrypeDB if there are none and raises
-        TooManyActiveGrypeDBs if more than one
+        Runs query against db to get active dbs and returns them in array
 
-        return: Active GrypeDBFeedMetadata
-        rtype: GrypeDBFeedMetadata
+        return: List of active GrypeDBFeedMetadata
+        rtype: List[GrypeDBFeedMetadata]
         """
-        try:
-            active_grypedb = cls._query_active_dbs()
-        except sqlalchemy.orm.exc.MultipleResultsFound:
-            logger.error("Too many active grype dbs found in db")
-            raise TooManyActiveGrypeDBs
+        return (
+            session.query(GrypeDBFeedMetadata)
+            .filter(GrypeDBFeedMetadata.active == True)
+            .order_by(GrypeDBFeedMetadata.created_at.desc())
+            .all()
+        )
 
-        if not active_grypedb:
+    @classmethod
+    def resolve_and_get_active_grypedb(cls, session: Session) -> GrypeDBFeedMetadata:
+        """
+        Public function to retrieve the current active grypedb in the database.
+        Raises NoActiveGrypeDB if no active db present
+        Resolves more than one active db by designating the most recent one as active and setting all others to inactive
+        *Note that this function may make commits on session object if there is more than one active db
+        """
+        active_dbs = cls._query_active_dbs(session)
+
+        # Raise error when no active db present
+        if len(active_dbs) == 0:
             logger.error("No active grype db found in the database")
             raise NoActiveGrypeDB
 
-        return active_grypedb
+        active_db = active_dbs[0]
 
-    @classmethod
-    def _query_active_dbs(cls) -> Optional[GrypeDBFeedMetadata]:
-        """
-        Runs query against db to get active dbs. Uses one_or_none so raises error if more than one active db
+        # If more than one active db, need to resolve correct active grypedb by setting outdated ones to inactive
+        if len(active_dbs) > 1:
+            for outdated_db in active_dbs[1:]:
+                outdated_db.active = False
 
-        return: Instance of GrypeDBFeedMetadata or None
-        rtype: GrypeDBFeedMetadata
-        """
-        db = get_thread_scoped_session()
-        return (
-            db.query(GrypeDBFeedMetadata)
-            .filter(GrypeDBFeedMetadata.active == True)
-            .one_or_none()
-        )
+            session.commit()
```

Comment:
@zburstein committing the session closes it, i don't that's correct here

Reply:
I thought commit closes the transaction, not the session

Reply:
No longer resolving through actually editing records